src/egraph.rs

```diff
@@ -833,8 +880,6 @@ impl<L: Language, N: Analysis<L>> EGraph<L, N> {
     ///
     /// When explanations are enabled, this function behaves like [`EGraph::union_trusted`],
     ///  and it lists the call site as the proof reason.
-    /// You should prefer [`union_instantiations`](EGraph::union_instantiations) when
```

Comment:
Don't remove this- `union_instantiations` is still preferred since it is the least confusing. 
However, you could add a note that alternatively, use `union_trusted` with uncanonical ids from `add_uncanonical`.